ant/windows/windows-packager.nsi.in

```diff
@@ -178,16 +174,20 @@ SectionEnd
 ;-------------------------------
 
 Function .onInit
+  ${If} ${RunningX64}
+      SetRegView 64
+  ${EndIf}
+  ${If} $InstDir == ""  
     ${If} ${RunningX64}
-        SetRegView 64
-        StrCpy $INSTDIR "$PROGRAMFILES64\${socket.name}"
+      StrCpy $INSTDIR "$PROGRAMFILES64\${socket.name}"
+    ${Else}
+      StrCpy $INSTDIR "$PROGRAMFILES\${socket.name}"
     ${EndIf}
+  ${EndIf}
 FunctionEnd
 
 Function un.onInit
     ${If} ${RunningX64}
         SetRegView 64
-        StrCpy $INSTDIR "$PROGRAMFILES64\${socket.name}"
```

Comment:
Yes, once i removed that line everything worked. The redirection is only there to make sure it removes the reg key from the correct location.